torch/autograd/functional.py

```diff
@@ -429,6 +436,42 @@ def jacobian(func, inputs, create_graph=False, strict=False):
     _check_requires_grad(outputs, "outputs", strict=strict)
 
     jacobian: Tuple[torch.Tensor, ...] = tuple()
+
+    if vectorize:
+        for outidx, out in enumerate(outputs):
+            flat_out = out.reshape(-1)
+
+            def vjp(grad_out):
+                vj = _autograd_grad((flat_out,), inputs, (grad_out,),
+                                    retain_graph=True, create_graph=create_graph)
+                vj = list(vj)
+                for el_idx, vj_el in enumerate(vj):
+                    if vj_el is not None:
+                        if strict and create_graph and not vj_el.requires_grad:
+                            msg = ("The jacobian of the user-provided function is "
+                                   f"independent of input {el_idx}. This is not allowed in "
+                                   "strict mode when create_graph=True.")
+                            raise RuntimeError(msg)
+                    else:
+                        if strict:
+                            msg = (f"Output {outidx} of the user-provided function is "
+                                   f"independent of input {el_idx}. This is not allowed in "
+                                   "strict mode.")
+                            raise RuntimeError(msg)
+                        vj[el_idx] = torch.zeros_like(inputs[el_idx])
+                return tuple(vj)
+
+            basis_vectors = torch.eye(flat_out.numel(), dtype=flat_out.dtype, device=flat_out.device)
```

Comment:
nit: we don't have a `eye_like` function?

Reply:
Unfortunately we don't